lib/open_uri.ml

```diff
@@ -29,7 +29,13 @@ let open_tcp f host port =
       let host_entry = Unix.gethostbyname host in
       let sockaddr = Unix.ADDR_INET(host_entry.Unix.h_addr_list.(0), port) in
       let s = Unix.socket Unix.PF_INET Unix.SOCK_STREAM 0 in
-      Unix.connect s sockaddr;
+      begin
+        try
+          Unix.connect s sockaddr
+        with e ->
+          Unix.close s;
+          raise e
+      end;
```

Comment:
Is there scope for wrapping L32-38 into a function and calling it in the rest of this PR?

Reply:
Why not. It could be a `handle_connect` defined next to `handle_socket`: https://github.com/xapi-project/xcp-idl/pull/225/files#diff-5240fdee46a08e096114e5144c900f2aL18

Reply:
Or, even better, we could actually move the connect into the `try` block of `handle_socket` itself

Reply:
yes